docs/cog_guides/dev.rst

```diff
@@ -0,0 +1,251 @@
+keywords, e.g. yield, will result in a syntax error. For multiple
+lines or asynchronous code, see [p]repl or [p]eval.
+
+**Environment Variables**
+
+* ``ctx``: Command invocation context
+* ``bot``: The bot object
+* ``channel``: The current channel object
+* ``author``: The current author's member object
+* ``guild``: The current guild object
+* ``message``: The command's message object
+* ``aiohttp``: The aiohttp library
+* ``asyncio``: The asyncio library
+* ``discord``: The discord.py library
+* ``commands``: The redbot.core.commands module
+* ``cf``: The redbot.core.utils.chat_formatting module
+* ``_``: The result from the last dev command
+
+**Arguments**
+
+* ``<code>``: The code to debug.
+
+.. _dev-command-eval:
+
+^^^^
+eval
+^^^^
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]eval <body>
+
+**Description**
+
+Execute asynchronous code.
+
+This command wraps code into the body of an async function and then
+calls and awaits it. The bot will respond with anything printed to
+stdout, as well as the return value of the function.
+
+The code can be within a codeblock, inline code or neither, as long
+as they are not mixed and they are formatted correctly.
+
+**Environment Variables**
+
+* ``ctx``: Command invocation context
+* ``bot``: The bot object
+* ``channel``: The current channel object
+* ``author``: The current author's member object
+* ``guild``: The current guild object
+* ``message``: The command's message object
+* ``aiohttp``: The aiohttp library
+* ``asyncio``: The asyncio library
+* ``discord``: The discord.py library
+* ``commands``: The redbot.core.commands module
+* ``cf``: The redbot.core.utils.chat_formatting module
+* ``_``: The result from the last dev command
+
+**Arguments**
+
+* ``<body>``: The code to evaluate.
+
+.. _dev-command-mock:
+
+^^^^
+mock
+^^^^
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]mock <user> <command>
+
+**Description**
+
+Mock another user invoking a command. The prefix must not be entered.
+
+**Arguments**
+
+* ``<member>``: The user to mock. |user-input-quotes|
+* ``<command>``: The command to invoke.
+
+.. _dev-command-mockmsg:
+
+^^^^^^^
+mockmsg
+^^^^^^^
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]mockmsg <user> <content>
+
+**Description**
+
+Dispatch a message event as if it were sent by a different user.
+
+Only reads the raw content of the message. Attachments, embeds etc. are
+ignored.
+
+**Arguments**
+
+* ``<user>``: The member to mock. |user-input-quotes|
+* ``<content>``: The content used for the message.
+
+.. note:: 
+
+        If ``content`` isn't passed, the message needs to contain embeds, attachments,
+        or anything else that makes the message non-empty.
+
+.. _dev-command-repl:
+
+^^^^
+repl
+^^^^
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]repl 
+
+**Description**
+
+Open an interactive REPL.
+
+The REPL will only recognise code as messages which start with a
+backtick. This includes codeblocks, and as such multiple lines can be
+evaluated.
```

Comment:
Though it is said when the repl is started, I think it would be helpful to include the fact that `exit` or `quit()` are used to exit the repl session, and possibly that those statements also need to start with a backtick to be interpreted. This could be added to the docstring as well.